tests/integration/targets/grafana_datasource/tasks/zabbix.yml

```diff
@@ -0,0 +1,126 @@
+- name: Create zabbix datasource
+  register: result
+  grafana_datasource:
+    name: datasource-zabbix
+    grafana_url: "{{ grafana_url }}"
+    grafana_user: "{{ grafana_username }}"
+    grafana_password: "{{ grafana_password }}"
+    org_id: '1'
+    ds_type: alexanderzobnin-zabbix-datasource
+    ds_url: https://zabbix.company.com
+    zabbix_user: grafana
+    zabbix_password: '******'
+
+- debug:
+    var: result
+
+- assert:
+    that:
+    - result.changed
+    - not result.datasource.isDefault
+    - result.datasource.jsonData.username == 'grafana'
+    - result.datasource.name == 'datasource-zabbix'
+    - result.datasource.orgId == 1
+    - result.datasource.password == ''
+    - result.datasource.type == 'alexanderzobnin-zabbix-datasource'
+    - result.datasource.url == 'https://zabbix.company.com'
+    - result.datasource.user == ''
+    - not result.datasource.withCredentials
+    - "result.msg == 'Datasource datasource-zabbix created'"
+
+- name: Create zabbix datasource (idempotency)
+  register: result
+  grafana_datasource:
+    name: datasource-zabbix
+    grafana_url: "{{ grafana_url }}"
+    grafana_user: "{{ grafana_username }}"
+    grafana_password: "{{ grafana_password }}"
+    org_id: '1'
+    ds_type: alexanderzobnin-zabbix-datasource
+    ds_url: https://zabbix.company.com
+    zabbix_user: grafana
+    zabbix_password: '******'
+
+- debug:
+    var: result
+
+- assert:
+    that:
+    - not result.changed
+    - not result.datasource.isDefault
+    - result.datasource.jsonData.username == 'grafana'
+    - result.datasource.name == 'datasource-zabbix'
+    - result.datasource.orgId == 1
+    - result.datasource.password == ''
+    - result.datasource.type == 'alexanderzobnin-zabbix-datasource'
+    - result.datasource.url == 'https://zabbix.company.com'
+    - result.datasource.user == ''
+    - not result.datasource.withCredentials
+
+- name: Update zabbix datasource
+  register: result
+  grafana_datasource:
+    name: datasource-zabbix
+    grafana_url: "{{ grafana_url }}"
+    grafana_user: "{{ grafana_username }}"
+    grafana_password: "{{ grafana_password }}"
+    org_id: '1'
+    ds_type: alexanderzobnin-zabbix-datasource
+    ds_url: https://zabbix.example.com
+    zabbix_user: grafana
+    zabbix_password: '******'
+
+- debug:
+    var: result
+
+- assert:
+    that:
+    - result.changed
+    - not result.datasource.isDefault
+    - result.datasource.jsonData.username == 'grafana'
+    - result.datasource.name == 'datasource-zabbix'
+    - result.datasource.orgId == 1
+    - result.datasource.password == ''
+    - result.datasource.type == 'alexanderzobnin-zabbix-datasource'
+    - result.datasource.url == 'https://zabbix.example.com'
+    - result.datasource.user == ''
```

Comment:
Is that expected or a bug ? User should not be empty right ?

Reply:
I am not sure, I copied the elastic test.

_result.datasource.jsonData.username_ has to match with _zabbix_user_. We don't need to check _result.datasource.user_ and _result.datasource.password_ for this example, maybe we can remove them.

Reply:
ok, it's not new then.